ibis/pyspark/compiler.py

```diff
@@ -0,0 +1,498 @@
+
+    def fn(col):
+        return ~F.max(col)
+    return compile_aggregator(t, expr, fn, context)
+
+
+@compiles(ops.All)
+def compile_all(t, expr, context=None):
+    return compile_aggregator(t, expr, F.min, context)
+
+
+@compiles(ops.NotAll)
+def compile_notall(t, expr, context=None):
+
+    def fn(col):
+        return ~F.min(col)
+    return compile_aggregator(t, expr, fn, context)
+
+
+@compiles(ops.Count)
+def compile_count(t, expr, context=None):
+    return compile_aggregator(t, expr, F.count, context)
+
+
+@compiles(ops.Max)
+def compile_max(t, expr, context=None):
+    return compile_aggregator(t, expr, F.max, context)
+
+
+@compiles(ops.Min)
+def compile_min(t, expr, context=None):
+    return compile_aggregator(t, expr, F.min, context)
+
+
+@compiles(ops.Mean)
+def compile_mean(t, expr, context=None):
+    return compile_aggregator(t, expr, F.mean, context)
+
+
+@compiles(ops.Sum)
+def compile_sum(t, expr, context=None):
+    return compile_aggregator(t, expr, F.sum, context)
+
+
+@compiles(ops.StandardDev)
+def compile_std(t, expr, context=None):
+    how = expr.op().how
+
+    if how == 'sample':
+        fn = F.stddev_samp
+    elif how == 'pop':
+        fn = F.stddev_pop
+    else:
+        raise AssertionError("Unexpected how: {}".format(how))
+
+    return compile_aggregator(t, expr, fn, context)
+
+
+@compiles(ops.Variance)
+def compile_variance(t, expr, context=None):
+    how = expr.op().how
+
+    if how == 'sample':
+        fn = F.var_samp
+    elif how == 'pop':
+        fn = F.var_pop
+    else:
+        raise AssertionError("Unexpected how: {}".format(how))
+
+    return compile_aggregator(t, expr, fn, context)
+
+
+@compiles(ops.Arbitrary)
+def compile_arbitrary(t, expr, context=None):
+    how = expr.op().how
+
+    if how == 'first':
+        fn = functools.partial(F.first, ignorenulls=True)
+    elif how == 'last':
+        fn = functools.partial(F.last, ignorenulls=True)
+    else:
+        raise NotImplementedError
+
+    return compile_aggregator(t, expr, fn, context)
+
+
+@compiles(ops.WindowOp)
+def compile_window_op(t, expr):
+    op = expr.op()
+    return t.translate(op.expr).over(compile_window(op.window))
+
+
+@compiles(ops.Greatest)
+def compile_greatest(t, expr):
+    op = expr.op()
+
+    src_columns = t.translate(op.arg)
+    if len(src_columns) == 1:
+        return src_columns[0]
+    else:
+        return F.greatest(*src_columns)
+
+
+@compiles(ops.Least)
+def compile_least(t, expr):
+    op = expr.op()
+
+    src_columns = t.translate(op.arg)
+    if len(src_columns) == 1:
+        return src_columns[0]
+    else:
+        return F.least(*src_columns)
+
+
+@compiles(ops.Abs)
+def compile_abs(t, expr):
+    op = expr.op()
+
+    src_column = t.translate(op.arg)
+    return F.abs(src_column)
+
+
+@compiles(ops.Round)
+def compile_round(t, expr):
+    op = expr.op()
+
+    src_column = t.translate(op.arg)
+    scale = op.digits.op().value if op.digits is not None else 0
```

Comment:
I think you want to translate this instead of pulling out the value from the ibis literal

Reply:
Fixed